### PR TITLE
SDK can be started without objects

### LINKED ISSUE #677: SDK should be started without objects, then added later
**Is this a BUG REPORT or FEATURE REQUEST?**:
BUG

**What happened**:
The SDK currently requires cluster and volume objects to be available before starting the SDK. 

**What you expected to happen**:
The SDK should be able to be started without objects, then cluster and volume objects can be added once. When the SDK is started without objects, it should return the appropriate helpful error to the caller when a function is called which requires that object.

### PR DESCRIPTION
**Pull request by lpabon**
**What this PR does / why we need it**:
This provides a way for the SDK to be started without cluster,
volume driver, or alerts. The respective handlers should check
if their necessary objects are available before continuing.

An object can be added to the SDK while running. For this to
work, the SDK has a rwlock interceptor, which is used to wait
until all requests are complete before adjust the object handlers.

**Which issue(s) this PR fixes** (optional)
Part of #677 

**Special notes for your reviewer**:
I tried to make sure during tests that we can catch any non-handled checks, but I could not figure out how to do this automatically. We will need to make sure as we add new handlers to check for objects, specially during reviews.

### CODE REVIEW

**Comment by lpabon** [comment]
This change needs to go in before #692

File: api/server/sdk/server.go [hunk 1]
```diff
@@ -58,7 +63,17 @@ type ServerConfig struct {
 type Server struct {
 	*grpcserver.GrpcServer
 
-	restPort             string
+	restPort            string
+	onceSetCluster      sync.Once
```
Comment (adityadani) [inline api/server/sdk/server.go]: sync.Once not used anywhere

File: api/server/sdk/server.go [hunk 2]
```diff
@@ -304,3 +377,16 @@ func (s *Server) restServerSetupHandlers() (*http.ServeMux, error) {
 
 	return mux, nil
 }
+
+// This interceptor provides a way to lock out any calls while we adjust the server
+func (s *Server) rwlockIntercepter(
```
Comment (adityadani) [inline api/server/sdk/server.go]: This is great!

File: api/server/sdk/server.go (hunk 1 above)
Comment (lpabon) [reply api/server/sdk/server.go]: oh, nice catch, i was trying different algorithms. I will remove it

**Pull request merged**